src/services/defectsService.js

```
@@ -0,0 +1,657 @@
+    /**
+     * Enhanced API response handler (following reference methodology)
+     * @param {Response} response - The fetch response object
+     * @returns {any} The extracted data
+     * @throws {Error} If the response indicates an error
+     */
+    async handleApiResponse(response) {
+      if (response.status >= 200 && response.status < 300) {
+        const data = await response.json();
+        
+        // Lambda function often returns data in the body property (like reference service)
+        if (data && data.body) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)